docs/core/component.md

```diff
@@ -244,12 +244,12 @@ the data to modify the entity. The handlers will usually interact with the
 
 Within the methods, we have access to the component prototype via `this`:
 
-| Property        | Description                                                                                                                                   |
-|-----------------|-----------------------------------------------------------------------------------------------------------------------------------------------|
-| this.data       | Parsed component properties computed from the schema default values, mixins, and the entity's attributes.                                     |
-| this.el         | Reference to the [entity][entity] as an HTML element.                                                                                         |
-| this.el.sceneEl | Reference to the [scene][scene] as an HTML element.                                                                                           |
-| this.id         | If the component can have [multiple instances][multiple], the ID of the individual instance of the component (e.g., `foo` from `sound__foo`). |
+| Property        | Description                                                                                                                                                                                                                                         |
+|-----------------|-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| this.data       | Parsed component properties computed from the schema default values, mixins, and the entity's attributes. <br/>__Note:__ The data property should be considered readonly. If a component wants to update its own data, it should use *setAttribute*.|
```

Comment:
Should we remove it all together from the docs to avoid temptations? 😄

Reply:
well, component authors need to know about the existence of `this.data`. Ideally we could solve this in code by using getters and setters and write-disabled properties but that's probably more work than it's worth.

Reply:
I was thinking that maybe the mention in the different methods documentation could be enough: https://aframe.io/docs/0.7.0/core/component.html#update-olddata
But yeah you also need to be aware of it for any other component method.

Reply:
What about:  
**Important:** Do not modify the data attribute directly. It is updated internally by A-Frame. To modify a component use *[setAttribute](https://aframe.io/docs/0.7.0/introduction/javascript-events-dom-apis.html#updating-a-component-with-setattribute)*

Reply:
👍 sounds good to me. I've pushed the change.